central/deployment/datastore/datastore_impl_postgres_test.go

```diff
@@ -0,0 +1,343 @@
+			orderMatters: false,
+			expectedIDs:  []string{dep2.Id},
+		},
+		{
+			desc:         "Search deployments with deployment scope and out-of-scope image query",
+			ctx:          scoped.Context(ctx, scoped.Scope{ID: dep2.Id, Level: v1.SearchCategory_DEPLOYMENTS}),
+			query:        pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.ImageOS, img3.GetScan().GetOperatingSystem()).ProtoQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{},
+		},
+		{
+			desc:         "Search deployments with image scope",
+			ctx:          scoped.Context(ctx, scoped.Scope{ID: img2.Id, Level: v1.SearchCategory_IMAGES}),
+			query:        pkgSearch.EmptyQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{dep2.Id},
+		},
+		{
+			desc:         "Search deployments with image scope and in-scope deployment query",
+			ctx:          scoped.Context(ctx, scoped.Scope{ID: img2.Id, Level: v1.SearchCategory_IMAGES}),
+			query:        pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.Namespace, dep2.GetNamespace()).ProtoQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{dep2.Id},
+		},
+		{
+			desc:         "Search deployments with image scope and out-of-scope deployment query",
+			ctx:          scoped.Context(ctx, scoped.Scope{ID: img2.Id, Level: v1.SearchCategory_IMAGES}),
+			query:        pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.Namespace, dep3.GetNamespace()).ProtoQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{},
+		},
+		{
+			desc: "Search deployments with image component scope",
+			ctx: scoped.Context(ctx, scoped.Scope{
+				ID: scancomponent.ComponentID(
+					img2.GetScan().GetComponents()[0].GetName(),
+					img2.GetScan().GetComponents()[0].GetVersion(),
+					img2.GetScan().GetOperatingSystem()),
+				Level: v1.SearchCategory_IMAGE_COMPONENTS,
+			}),
+			query:        pkgSearch.EmptyQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{dep2.Id},
+		},
+		{
+			desc: "Search deployments with image vuln scope",
+			ctx: scoped.Context(ctx, scoped.Scope{
+				ID: cve.ID(
+					img1.GetScan().GetComponents()[0].GetVulns()[0].GetCve(),
+					img1.GetScan().GetOperatingSystem()),
+				Level: v1.SearchCategory_IMAGE_VULNERABILITIES,
+			}),
+			query:        pkgSearch.EmptyQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{dep1.Id, dep3.Id},
+		},
+		{
+			desc:         "Search images with empty query",
+			ctx:          ctx,
+			query:        pkgSearch.EmptyQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{img1.Id, img2.Id, img3.Id},
+			queryImages:  true,
+		},
+		{
+			desc:         "Search images with deployment query",
+			ctx:          ctx,
+			query:        pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.Namespace, dep2.GetNamespace()).ProtoQuery(),
+			orderMatters: true,
+			expectedIDs:  []string{img2.Id},
+			queryImages:  true,
+		},
+		{
+			desc:         "Search images with deployment+namespace query",
```

Comment:
image + namespace?

Reply:
Same for the ones below. Where is deployment in the query?

Reply:
ah, I copied my tc pattern from one of my other branches.